src/plugins/intel_cpu/src/edge.cpp

```diff
@@ -52,22 +52,45 @@ bool Edge::isDropped() const {
 }
 
 void Edge::collectConsumers(std::vector<NodePtr>& result) const {
-    if (!this->getChild()->getChildEdges().empty() && this->inPlace(LOOK_DOWN)) {
-        if (auto peerChildSPD = this->getChild()->getSelectedPrimitiveDescriptor()) {
+    auto add_result_node = [](std::vector<NodePtr>& result, const NodePtr& node) -> bool {
+        if (Type::ShapeOf == node->getType()) {
+            // ShapeOf doesn't actually read the data, it only reads shape
+            return true;
```

Comment:
May be it's more logical to return `true` when the node has been added to the result vector?

Reply:
done